runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -220,7 +220,11 @@ public void rollbackUnsafe(long version) {
                     .doUndo(object, undoRecord.getUndoRecord(),
                                     undoRecord.getSMRArguments());
 
-            this.version = undoRecord.getEntry().getGlobalAddress();
+            // After undoing the record, we should get back to the previous version
+            // We use the backpointerMap to get the previous entry version
+            // If none found, throw NoRollbackException.
+            this.version = undoRecord.getEntry().getBackpointerMap().values().stream().
```

Comment:
I think you want

getBackpointerMap().get(sv.getID())

Which will produce the correct backpointer